src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -19,6 +19,7 @@
     <DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.1'">$(DefineConstants);EVENTSOURCE_ACTIVITY_SUPPORT</DefineConstants>
     <DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.1' and '$(TargetFramework)' != 'netstandard1.3'">$(DefineConstants);EVENTSOURCE_ENUMERATE_SUPPORT</DefineConstants>
     <DefineConstants Condition="$(TargetFramework.StartsWith('net4'))">$(DefineConstants);ALLOW_PARTIALLY_TRUSTED_CALLERS;ENABLE_HTTP_HANDLER</DefineConstants>
+    <DefineConstants Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">$(DefineConstants);NET5</DefineConstants>
```

Comment:
What happens in the future when NetCoreAppCurrent is > 5? Do we need to add netcoreapp5.0 to the TargetFrameworks list explicitly?

Reply:
> Do we need to add netcoreapp5.0 to the TargetFrameworks list explicitly?

No

> What happens in the future when NetCoreAppCurrent is > 5? 

NetCoreAppCurrent will have the right next version of netcoreapp. so in 6.0 will have it as 6.0. in short this should work moving forward. I chose the defined name as `NET5` to be clear when we have introduced this change when looking at the code.

Reply:
I think @ericstj raised a similar concern. The scenario I am worried about is that this NuGet package is distributed as an OOB package. Assume in 2021 we ship a new version that targets netcoreapp6 and people keeping their netcoreapp5.0 app upgrade this package only. I assume that NuGet will resolve the reference to the netstandard implementation because the updated version no longer targets netcoreapp5.0. The netstandard implementation wouldn't have default W3C id so the upgrade will will cause them to lose the change. I'm no expert in NuGet resolution so if I am misunderstanding the mechanics here just let me know : )

Reply:
Yes, you are right. I am talking to @ericstj if we need to include netcoreapp targeting to the package to mitigate this issue. 
CC @joperezr

Reply:
You've got it right.  We need to include the `net5.0` build in the package.  We might also remove the `netstandard2.0` build from this project.  I don't see anywhere it would actually be used.  We don't ship it in the package and it's not used in any other scenario.

Reply:
> We need to include the net5.0 

does his can be done by just removing the following line? or you think we need to do something else? 

```XML
<ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage> 
```

> We might also remove the netstandard2.0

I can do that too if you think it is safe to do so.

Reply:
Yeah, ideally we could do it in a way so that when we change the value of $(NetCoreAppCurrent) (eg net5.1) we will keep the net5.0 configuration in the package.  @joperezr or @Anipik: any suggestions?

Reply:
ok, thanks @ericstj. I have done that. please have a quick look at the last commit just to confirm the change.

Reply:
@ericstj could you please have a quick look at last commit and let me know? I want to ensure we are good to go.

Reply:
Actually I am seeing tons of failures like 
```
##[error].dotnet/sdk/5.0.100-preview.6.20310.4/Microsoft.Common.CurrentVersion.targets(2084,5): error MSB3245: (NETCORE_ENGINEERING_TELEMETRY=Build) Could not resolve this reference. Could not locate the assembly "System.Memory". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
/__w/1/s/.dotnet/sdk/5.0.100-preview.6.20310.4/Microsoft.Common.CurrentVersion.targets(2084,5): error MSB3245: Could not resolve this reference. Could not locate the assembly "System.Memory". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/__w/1/s/src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj]
```

https://dev.azure.com/dnceng/public/_build/results?buildId=693198&view=logs&j=5ea74393-5ab8-5088-11c3-b14699e9d5af&t=c9c7c300-3b25-5026-7590-be769f231fb8&l=5751

Reply:
Yeah that is because we actually need to keep ns2.0 configuration as it is used for Build From Source. In Build from source, apart from building the regular vertical, we also build the netstandard 2.0 vertical: https://github.com/dotnet/runtime/blob/master/src/libraries/Directory.Build.props#L288 and for that we were building the ns2.0 config on this project, but now the one that applies is ns1.3 which you can't build in build from source as we don't have the full ns1.3 targeting pack.

Reply:
I returned the netstandard 2.0 back for now.